webapp/src/lib/utils/time-table-util.ts

```diff
@@ -0,0 +1,71 @@
+import { DAY_IN_A_WEEK, TOTAL_SEGMENTS_NUMBER } from '$lib/constants';
+import { TimeSlotStatus } from '$lib/enum/timeslot-status';
+import type { components } from '$lib/sdk/schema';
+
+type TimeUnit = components['schemas']['TimeUnit'];
+
+export function toTimeUnits(map: Map<number, TimeSlotStatus[][]>): TimeUnit[] {
+    const result = new Array<TimeUnit>();
+    for (const [week, value] of map.entries()) {
+        const weekDate = new Date(week);
+        result.push(
+            ...value.flatMap((infoday, segment) =>
+                infoday.reduce((acc, status, day) => {
+                    if (status !== TimeSlotStatus.AVAILABLE) {
+                        return acc;
+                    }
+                    const startTime = new Date(weekDate);
+                    startTime.setDate(weekDate.getDate() + day);
+                    startTime.setHours(0, segment * 30, 0, 0);
+                    const endTime = new Date(startTime);
+                    endTime.setMinutes(endTime.getMinutes() + 30);
+                    acc.push({
+                        start_time: startTime.toISOString(),
+                        end_time: endTime.toISOString()
+                    });
+                    return acc;
+                }, new Array<TimeUnit>())
+            )
+        );
+    }
+    return result;
+}
+
+export function initializeAvailabilityTables(
+    availabilityTables: Map<number, TimeSlotStatus[][]>,
+    current_monday: Date,
+    today: Date
+) {
+    for (
+        let week = new Date(current_monday);
+        week.getTime() < today.getTime();
+        week.setDate(week.getDate() + 7)
+    ) {
+        const availability: TimeSlotStatus[][] = new Array(TOTAL_SEGMENTS_NUMBER);
+        availabilityTables.set(week.getTime(), availability);
+
+        for (
+            let day = new Date(week);
+            day.getTime() < today.getTime();
+            day.setDate(day.getDate() + 1)
+        ) {
+            let total_segment_number = TOTAL_SEGMENTS_NUMBER;
+            const midnightToday = new Date(today);
```

Comment:
Can we rename the variables to be both in CamelCase or with _?

Reply:
javaScript convention use CamelCase so I will change all variable to CamelCase